openquake/hazardlib/calc/filters.py

```diff
@@ -86,6 +87,38 @@ def filter_sites_by_distance_to_rupture(rupture, integration_distance, sites):
     return sites.filter(jb_dist <= integration_distance)
 
 
+def filtered_site_to_rupture_distance(rupture, integration_distance, sites):
+    """
+    Filter out sites from the collection that are further from the rupture
+    than some arbitrary threshold.
+
+    :param rupture:
+        Instance of :class:`~openquake.hazardlib.source.rupture.Rupture`
+        that was generated by :meth:
+        `openquake.hazardlib.source.base.BaseSeismicSource.iter_ruptures`
+        of an instance of this class.
+    :param integration_distance:
+        Threshold distance in km.
+    :param sites:
+        Instance of :class:`openquake.hazardlib.site.SiteCollection`
+        to filter.
+    :returns:
+        Filtered :class:`~openquake.hazardlib.site.SiteCollection`.
+
+    This function is similar to :meth:`openquake.hazardlib.source.base.BaseSeismicSource.filter_sites_by_distance_to_source`.
+    The same notes about filtering criteria apply. Site
+    should not be filtered out if it is not further than the integration
+    distance from the rupture's surface projection along the great
+    circle arc (this is known as Joyner-Boore distance, :meth:`
+    openquake.hazardlib.geo.surface.base.BaseQuadrilateralSurface.get_joyner_boore_distance`).
+    """
+    jb_dist = rupture.surface.get_joyner_boore_distance(sites.mesh)
+    idx = jb_dist <= integration_distance
+    dctx = DistancesContext()
+    dctx.rjb = jb_dist[idx]
+    return sites.filter(idx), dctx
+
+
```

Comment:
I do not see this function called in the branch ucerf of oq-risklib, so it looks like we do not need it.